pkg/acquisition/modules/loki/loki.go

```diff
@@ -0,0 +1,388 @@
+	params := &url.Values{}
+	params.Set("query", l.Config.Query)
+	params.Set("direction", "forward")
+	params.Set("since", time.Time(l.Config.Since).Format(time.RFC3339))
+	params.Set("limit", "1000") // FIXME
+	url := fmt.Sprintf("%s/loki/api/v1/query_range?%s",
+		l.Config.URL,
+		params.Encode())
+	logger := l.logger.WithField("url", url)
+	req, err := http.NewRequest("GET", url, nil)
+	req.Header = l.header
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		logger.WithError(err).Error("http error")
+		return errors.Wrap(err, "Error while querying loki")
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		msg, _ := io.ReadAll(resp.Body)
+		logger.WithField("status", resp.StatusCode).Error(string(msg))
+		return fmt.Errorf("Loki query return bad status : %d", resp.StatusCode)
+	}
+	var lq LokiQuery
+	/*
+		fmt.Println(url)
+		msg, _ := io.ReadAll(resp.Body)
+		fmt.Println(string(msg))
+	*/
+	decoder := json.NewDecoder(resp.Body)
+	err = decoder.Decode(&lq)
+	if err != nil {
+		return errors.Wrap(err, "can't parse JSON loki response")
+	}
+	for _, result := range lq.Data.Result {
+		logger.WithField("stream", result.Stream).Debug("Results", len(result.Values))
+		for _, entry := range result.Values {
+			l.readOneEntry(entry, result.Stream, out)
+		}
+	}
+	logger.Info("Loki queried")
+	t.Kill(nil)
+	return nil
+}
+
+func (l *LokiSource) readOneTail(resp Tail, out chan types.Event) {
+	for _, stream := range resp.Streams {
+		for _, entry := range stream.Entries {
+			l.readOneEntry(entry, stream.Stream, out)
+		}
+	}
+}
+
+func (l *LokiSource) readOneEntry(entry Entry, labels map[string]string, out chan types.Event) {
+	ll := types.Line{}
+	ll.Raw = entry.Line
+	ll.Time = entry.Timestamp
+	ll.Src = l.Config.URL
+	ll.Labels = labels
+	ll.Process = true
+	ll.Module = l.GetName()
+
+	linesRead.With(prometheus.Labels{"source": l.Config.URL}).Inc()
+	out <- types.Event{
+		Line:       ll,
+		Process:    true,
+		Type:       types.LOG,
+		ExpectMode: leaky.TIMEMACHINE,
+	}
+}
+
+func (l *LokiSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	err := l.ready()
+	if err != nil {
+		return errors.Wrap(err, "error while getting StreamingAcquisition")
+	}
+	t.Go(func() error {
+		for {
+			ctx, cancel := context.WithTimeout(context.TODO(), readyTimeout)
+			defer cancel()
+			c, res, err := l.dialer.DialContext(ctx, l.lokiWebsocket, l.header)
+			if err != nil {
+				buf, err2 := ioutil.ReadAll(res.Body)
+				if err2 == nil {
+					return fmt.Errorf("loki websocket (%s) error %v : %s", l.lokiWebsocket, err, string(buf))
+				}
+				return err2
+			}
+			defer c.Close()
+			var resp Tail
+			for { // draining the websocket
+				t, msg, err := c.ReadMessage()
+				if len(msg) == 0 {
+					time.Sleep(100 * time.Millisecond)
+					continue
+				}
+				err = json.Unmarshal(msg, &resp)
+				if err != nil {
+					return errors.Wrap(err, "OneShotAcquisition error while reading JSON websocket")
```

Comment:
StreamingAcquisition instead of OneShotAcquisition

Reply:
Oups, an ugly copy/paste error.

See d36ea8f48fa71ea6bfa1f2f98249a0bcd7a8d576